net/respondd/src/respondd.c

```diff
@@ -388,15 +547,21 @@ int main(int argc, char **argv) {
 		exit(EXIT_FAILURE);
 	}
 
+	if (setsockopt(sock, IPPROTO_IPV6, IPV6_RECVPKTINFO, &one, sizeof(one))) {
+		perror("can't set socket to deliver IPV6_PKTINFO control message");
+		exit(EXIT_FAILURE);
+	}
+
 	server_addr.sin6_family = AF_INET6;
 	server_addr.sin6_addr = in6addr_any;
 
 	opterr = 0;
 
 	int group_set = 0;
+	int max_multicast_delay = 10000;
```

Comment:
I'm not sure about the best value for the maximum delay. The current test does not say anything about the behavior in the real scenario.

Some more facts about the graph:
* I have deployed the patch to a few routers (experimental branch) in the freifunk hanover community and the remaining routers are still answering to the multicast request immediately. \*
* The guy, who built the experimental firmware wanted to use a greater delay, so the test was done with cli option ```-t 60```. 

I do not know, how to optimize the delay, since this should be done in a more realistic scenario. I don't want to deploy the patch again and again to all nodes in my community.

-

\* Maybe some of the remaining loss is caused by this environment. If the random delay (of the patched router) is very short, the response may still conflict with the bulk of responses of the unpatched ones.